text/0000-lifecycle-all.md

```diff
@@ -0,0 +1,110 @@
+
+Most of the these flags are existing lifecycle phase flags. `-skip-restore` is a a new addition that modifies the behavior of `analyzer` similarly to the existing `-skip-layers` flag, and, in addition, skips the `restorer` phase all together.
+
+This RFC proposes a `-tag` flag as the interface to provide additional tags, rather than multiple positional arguments (as is currently accepted by `exporter` ) for clarity and it's similarity to `docker build`.
+
+Some existing lifecycle flags (e.g. `-group` on the `detector` `builder` and `exporter` phases) are not necessary. They refer to files that were previously used to pass information between lifecycle phase processes. When the phases run in the same process, those values can be passed in memory.
+
+
+### User
+
+Right now, when `pack` runs w/o the `--publish` flag `analyzer` and `exporter` are run as root. This is required so that these phases can connect to the mounted docker daemon socket. Historically the `builder` and `detector` binaries are never run as root.
+
+In the daemon case pack will run `all` as `root` but the buildpacks' `/bin/detect` and `/bin/builder` scripts will be invoked as the provided user. This will require a change to build and detect implementations.
+
+### Credential Management
+
+Right now, when `pack` runs w/ the `--publish` flag `analyzer` and `exporter` are provided with registry credentials via the `CNB_REGISTRY_AUTH` environment variable.
+
+In the registry case pack will set `CNB_REGISTRY_AUTH` when invoking `all`. To prevents buildpacks from having read access to those credentials the build and detect implementations will ensure this variable is not present in the environment of the `/bin/detect` and `/bin/build` processes.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+Making it easy to run all of the lifecycle in a single container will encourage folks to do so. However, this requires consumers to be more thoughtful about security and credential isolation. We will need to harden `builder` and `detector` implementation to ensure buildpacks are not given root access or registry credentials.
+
+If we choose to implement [rfcs#43](https://github.com/buildpacks/rfcs/pull/43) `pack` will only accrue performance improvements from this change when a builder is explicitly trusted. In the untrusted case pack must continue executing each phase in a separate container.
+
+# Alternatives
+[alternatives]: #alternatives
+
+Require platforms/integrations to invoke all lifecycle phases in order in all situations, even when running in a single container.
+
+# Prior Art
+[prior-art]: #prior-art
+
+Most in container build tools do not spin up multiple containers.
+
+Regarding the introduction of the `-tag` flag, `docker build -t` is prior art.
+
+# Unresolved Questions
+[unresolved-questions]: #unresolved-questions
+
+- Is there a better name than `lifecycle/all`? It doesn't include `launcher` or `rebaser` functionality so it isn't truly `all`.
```

Comment:
* `lifecycle/accio` 
* `lifecycle/alohomora` 
* `lifecycle/morsmordre` (dark magic)
* `lifecycle/sectumsempra` (very dark magic)

Reply:
```
lifecycle/sectumsempra
```
lifecycle/silencio

Reply:
`lifecycle/Ash_nazg` to rule them all